Sources/Datadog/Logging/RemoteLogger.swift

```diff
@@ -0,0 +1,144 @@
+
+    /// Logger-specific attributes (must be accesset through `queue`).
+    private var unsafeAttributes: [String: Encodable] = [:]
+    /// Logger-specific tags (must be accesset through `queue`).
+    private var unsafeTags: Set<String> = []
+    /// Queue for ensuring thread-safety of tags and attributes mutation.
+    private let queue: DispatchQueue
+
+    /// Date provider for logs.
+    private let dateProvider: DateProvider
+    /// Builds log events.
+    private let builder: LogEventBuilder
+    /// Integration with RUM Context. `nil` if disabled for this logger or if the RUM feature disabled.
+    internal let rumContextIntegration: LoggingWithRUMContextIntegration?
+    /// Integration with Tracing. `nil` if disabled for this logger or if the Tracing feature disabled.
+    internal let activeSpanIntegration: LoggingWithActiveSpanIntegration?
+
+    init(
+        core: DatadogCoreProtocol,
+        configuration: Configuration,
+        dateProvider: DateProvider,
+        rumContextIntegration: LoggingWithRUMContextIntegration?,
+        activeSpanIntegration: LoggingWithActiveSpanIntegration?
+    ) {
+        self.core = core
+        self.configuration = configuration
+        self.dateProvider = dateProvider
+        self.queue = DispatchQueue(
+            label: "com.datadoghq.logger-\(configuration.loggerName)",
+            target: .global(qos: .userInteractive)
+        )
+        self.builder = LogEventBuilder(
+            service: configuration.service,
+            loggerName: configuration.loggerName,
+            sendNetworkInfo: configuration.sendNetworkInfo,
+            eventMapper: configuration.eventMapper
+        )
+        self.rumContextIntegration = rumContextIntegration
+        self.activeSpanIntegration = activeSpanIntegration
+    }
+
+    // MARK: - Attributes
+
+    func addAttribute(forKey key: AttributeKey, value: AttributeValue) {
+        queue.async { self.unsafeAttributes[key] = value }
+    }
+
+    func removeAttribute(forKey key: AttributeKey) {
+        queue.async { self.unsafeAttributes.removeValue(forKey: key) }
+    }
+
+    // MARK: - Tags
+
+    func addTag(withKey key: String, value: String) {
+        queue.async { self.unsafeTags.insert("\(key):\(value)") }
+    }
+
+    func removeTag(withKey key: String) {
+        queue.async { self.unsafeTags = self.unsafeTags.filter { !$0.hasPrefix("\(key):") } }
+    }
+
+    func add(tag: String) {
+        queue.async { self.unsafeTags.insert(tag) }
+    }
+
+    func remove(tag: String) {
+        queue.async { self.unsafeTags.remove(tag) }
+    }
+
+    // MARK: - Logging
+
+    func log(level: LogLevel, message: String, error: Error?, attributes: [String: Encodable]?) {
+        guard level.rawValue >= configuration.threshold.rawValue else {
+            return
+        }
+
+        // on public API caller thread:
+        let date = dateProvider.now
+        let threadName = getCurrentThreadName()
```

Comment:
We record `date` and `threadName` before jumping off from the caller thread. Alternatively, the `date` could be read from SDK `context`, but then it would require `queue.sync {}` to manage logger tags and attributes ([like we do in V1](https://github.com/DataDog/dd-sdk-ios/blob/f330ac00db328e110ca5287818c58cc9aaa706ec/Sources/Datadog/Logger.swift#L266-L270)). I made an extra annotation on `date` in `LogEventBuilder.createLogEvent()` to make it documented and clear.

Reply:
Yes, lets keep the date outside of the context thread. It will be the same behaviour with v2 context 👍